features/step_definitions/workflow.js

```diff
@@ -0,0 +1,208 @@
+'use strict';
+
+const Assert = require('chai').assert;
+const github = require('../support/github');
+const request = require('../support/request');
+const sdapi = require('../support/sdapi');
+const { defineSupportCode } = require('cucumber');
+
+const TIMEOUT = 240 * 1000;
+
+defineSupportCode(({ Before, Given, When, Then }) => {
+    Before({
+        tags: '@workflow'
+    }, function hook() {
+        this.repoOrg = this.testOrg;
+        this.repoName = 'functional-workflow';
+        this.pipelineId = null;
+        this.builds = null;
+    });
+
+    Given(/^an existing pipeline on "(.*)" branch with the workflow jobs:$/, {
+        timeout: TIMEOUT
+    }, function step(branch, table) {
+        return this.getJwt(this.apiToken)
+            .then((response) => {
+                this.jwt = response.body.token;
+
+                return github.createBranch(this.gitToken, branch, this.repoOrg, this.repoName);
+            })
+            .then(() => this.createPipeline(this.repoName, branch))
+            .then((response) => {
+                Assert.oneOf(response.statusCode, [409, 201]);
+
+                if (response.statusCode === 201) {
+                    this.pipelineId = response.body.id;
+                } else {
+                    const str = response.body.message;
+                    const id = str.split(': ')[1];
+
+                    this.pipelineId = id;
+                }
+
+                return this.getPipeline(this.pipelineId);
+            })
+            .then((response) => {
+                const expectedJobs = table.hashes();
+
+                this.jobs = response.body;
+
+                for (let i = 0; i < expectedJobs.length; i += 1) {
+                    const job = this.jobs.filter(j => j.name === expectedJobs[i].job);
```

Comment:
We should use `find()` function instead of `filter()`, and then we can omit `[0]` of  the `job[0]`.

Reply:
Thanks, fixed that.